File: frontend/src/types/diagnostics.ts
export type DiagnosticStatus = 'ok' | 'warning' | 'error'

export interface DiagnosticCheck {
  name: string
  status: DiagnosticStatus
  message: string
}

export interface DiagnosticsResponse {
  message: string
  checks: DiagnosticCheck[]
}

File: backend/app/schemas/diagnostics.py
from typing import Literal

from pydantic import BaseModel

DiagnosticStatus = Literal["ok", "warning", "error"]


class DiagnosticCheck(BaseModel):
    name: str
    status: DiagnosticStatus
    message: str


class DiagnosticsResponse(BaseModel):
    message: str
    checks: list[DiagnosticCheck]
